rfc-005-template-contributing-process.md

```diff
@@ -0,0 +1,154 @@
+
+4. **Workshop the drafts with the community** - Revise and refine their template set with the community pull reviewers. At least three community reviews are required to move to the next phase.
+
+5. **Submit a pull request** - They open a pull request and it is reviewed for quality by the pull request reviewers. At least two pull request reviewer approvals are needed to be merged into the repository.
+
+6. **Improve the template with user feedback** - As users try the template in the wild, they may report usability issues or provide feedback for improvements to the template. Either the original template author or another templateer will evaluate feedback and incorporate it into future versions of the template. When that happens, the template will go through the previous phases again.
+
+Each template set will include:
+- **Raw template** - The basic template for the type of document you are creating. It provides a rough outline of the suggested content and a few embedded writing tips for how to fill in the different sections of the template.
+- **Template guide** - Provides a deeper explanation of how to fill in the raw template. It provides a lightweight introduction to the purpose of this documentation and explains how to fill in each section of the document. Additional content may be included as needed.
+- **Before you start (optional)** - This optional guide is designed to help template users understand some of the key research, brainstorming, or pre-writing steps that they might need to do before adding this type of document to a guide. If this documentation type has certain key decisions that need to be made with stakeholders or if it has a dependency on some other type of document or process, that should be mentioned in this guide.
+- **Template example (optional)** - The Good Docs Project is in the process of designing a fake documentation project that can provide examples of our templates in action. Until this project is ready, template authors may optionally create an example of the template in action.
+
+
+## Links and prior art
+
+If you'd like to view the comment and revision history of these documents, you can see them in their Google Doc form:
+
+- [Template contribuing guide](https://docs.google.com/document/d/19pLi0JMlib11g5Di076hjyUM5KRkPHPvZLQpCwmPYi8/edit?usp=sharing)
+- [Template deliverables](https://docs.google.com/document/d/1uZPQ6jSj0TQoFSII_q6SbHgSySaGAZAevLvTuvXJeEU/edit?usp=sharing)
+- [Template roles and resources](https://docs.google.com/document/d/1fXp0Q7mv-VFj9GxScoXG1ZahsaAnkI3drPxuVNf9JVQ/edit?usp=sharing)
+
+
+## Open questions
+
+We haven't really found a good title for the template deliverable I'm currently calling "Before you start." Cameron prefers "theory." Suggestions are appreciated! Other ideas that were floated in the community:

```

Comment:
+1 for 'before you start': very clear and helpful!

Reply:
I like Before You Start as well...

Reply:
thought (non-blocking): My reservations with "before you start" is that while this theory type doc is likely to be read before starting on a doc, I anticipate it will often be read as part of a **future** doc iteration, like a phase 2 improvement.

Another word I've recently [seen in the wild](https://www.w3.org/TR/skos-primer/) is a "primer". A conceptual document which provides the background theory related to a topic. While a primer need not be read "before you start" is does imply you should read first (like a primer coat of paint, being the first coat of paint you use). How would you feel about "primer"?

https://github.com/thegooddocsproject/templates/blob/dev/base/base-theory.md

Reply:
I do like primer. That could possibly work well because it doesn't sound as heavy as "theory," doesn't pre-suppose when you will read the doc, and doesn't make you instantly forecast what the content is going to be. Other reactions?

Reply:
I like `primer` as well.

Reply:
I prefer "About this Template" or just "About". Filename could be ***-about.md. I find Before you Start problematic because in many cases content like references supporting the decisions made to construct the template are simply not of interest to end users, and so it is incorrect to say that they need to review them before they use the template. Some users will want this, and perhaps even more common will be new templateers interested in deciding for themselves if they agree with the approach taken when creating the template originally.  Essential facts that the user should know before filling out the template should be in ***-guide, not in some other file. Primer also seems incorrect here --- it refers to an introductory educational book about a subject. The scope of what's being described is too small for primer to make sense. My understanding also is that knowledge essential to read before using the template should go in ****-guide. There seems to be quite a bit of overlap here about what the ***-guide is vs. ****-theory/before-you-start. I would rate this a blocking concern.

Reply:
Out of the already proposed names, I vote for “Template background”. But I would also like to suggest “Backdrop” or “Groundwork” implying just that. Because this is likely to contain all the information that went into setting the stage for the actual template.

Reply:
I don't have a strong opinion here, but "Getting Started" is a common term used for this or something like it, and people will know what it means.

Reply:
@aidanatwork , I see your point, but remember that the description for this content is "This optional guide is designed to help template users understand some of the key research, brainstorming, or pre-writing steps that they might need to do before adding this type of document to a guide. If this documentation type has certain key decisions that need to be made with stakeholders or if it has a dependency on some other type of document or process, that should be mentioned in this guide."

For that reason, I don't feel like "about" accurately describes what this document does.

For some extra context, the purpose of this additional optional document was discussed heavily in the Community Docs working group on August 8 and this was the decision that the 5 templateers in attendance gave. (See [Community Docs notes](https://docs.google.com/document/d/1iRxB0T7FoVjIeG_LawbCzcLfXCILlpFdzn-4BPDGR9Y/edit?usp=sharing) for more context.) Here's the key takeaways:

- Re: the base template "theory" document: The general recommendation was to include most of this material in a simpler, more lightweight form in the template guide document. The template guide document could open with a section about “Why do I need this type of document”, then it could give a guide for filling in all the sections of the document, then it could close with a list of links or references for deeper learning and research (including any Good Docs Project blog entries or white papers).
- We also had a brief conversation about content that seems to be missing: what about a guide that helps you do “pre-writing” and brainstorming or work through some of the research you need to do to prepare to write the template? Perhaps we could create a simple guide that asks you questions and gets you to think through what needs to be considered before you begin writing. This content could possibly go into the template guide (but that guide is starting to have a lot of content, isn’t it?) or it could be a separate document.

That second comment forms the basis for why it was originally called "before you start." It was meant to include pre-writing activities or questions to help you do the necessary groundwork you needed to do before you even sat down to write.

For that reason, I am still somewhat partial to "Getting Started" or "Before You Start" but I am also open to "Primer."

Reply:
Thanks to everyone who commented on the naming and content of the "before you start" document. We discussed this at length at last week's general meeting and the group consensus there was to call this the "deep dive" document. It basically contains miscellaneous information or guidance that is too comprehensive for the "guide" document. Based on whatever the each unique template needs, this document could include research, deep theories, pre-writing information, and/or any other material that users need to know when implementing this topic. Template writers are encouraged to point/link readers to that document in their guide to encourage users to read it in more depth.

I'm adding this content to the template documents and, when that is done, I will resolve these comments.